src/DDTrace/Integrations/OpenAI/OpenAIIntegration.php

```diff
@@ -0,0 +1,1220 @@
+                            $clientData = ObjectKVStore::get($transporter, 'client_data');
+                            ObjectKVStore::put($this, 'client_data', $clientData);
+                        }
+                        /** @var array{baseUri: string, headers: string, apiKey: ?string} $clientData */
+                        OpenAIIntegration::handleRequest(
+                            span: $span,
+                            methodName: $methodName,
+                            args: $args,
+                            basePath: $clientData['baseUri'],
+                            apiKey: $clientData['apiKey'],
+                        );
+                    },
+                    'posthook' => function (\DDTrace\SpanData $span, $args, $response) use ($logger, $httpMethod, $endpoint) {
+                        /** @var (\OpenAI\Contracts\ResponseContract&\OpenAI\Contracts\ResponseHasMetaInformationContract)|string $response */
+                        // Files::download - i.e., downloadFile - returns a string instead of a Response instance
+                        OpenAIIntegration::handleResponse(
+                            span: $span,
+                            logger: $logger,
+                            headers: $response ? (method_exists($response, 'meta') ? $response->meta()->toArray() : []) : [],
+                            response: \is_string($response) ? $response : ($response ? $response->toArray() : []),
+                            httpMethod: $httpMethod,
+                            endpoint: $endpoint,
+                        );
+                    }
+                ]
+            );
+        }
+
+        foreach ($streamedTargets as [$class, $method, $methodName, $httpMethod, $endpoint]) {
+            \DDTrace\trace_method(
+                $class,
+                $method,
+                [
+                    'prehook' => function (\DDTrace\SpanData $span, $args) use ($methodName, $integration) {
+                        $integration->setServiceName($span);
+                        $clientData = ObjectKVStore::get($this, 'client_data');
+                        if (\is_null($clientData)) {
+                            $transporter = ObjectKVStore::get($this, 'transporter');
+                            $clientData = ObjectKVStore::get($transporter, 'client_data');
+                            ObjectKVStore::put($this, 'client_data', $clientData);
+                        }
+                        /** @var array{baseUri: string, headers: string, apiKey: ?string} $clientData */
+                        OpenAIIntegration::handleRequest(
+                            span: $span,
+                            methodName: $methodName,
+                            args: $args,
+                            basePath: $clientData['baseUri'],
+                            apiKey: $clientData['apiKey'],
+                            streamed: true
+                        );
+                    },
```

Comment:
nit: Almost all this (except `streamed`) is duplicated between streamed/non-streamed. May not even be worth to simplify, though